[Sun Ze Dong] iP

zedonggg opened the pull request:
# Aik Hsu
A simple command-line chatbot for managing your tasks. (_~Named after my friend as a joke~_ 🍆)
> 🍆

## Features
- Manage events
- Manage deadlines
- Manage todos

## Coming soon
- [x] Tasks support
- [ ] Reminders
- [ ] Automations

## Installation
1. Download the latest **Aikhsu.jar** release from [here](https://github.com/zedonggg/ip/releases)
2. Double click to run
3. Follow the command-line instructions to manage your tasks

##
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
        new Aikhsu("Aikhsu.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zedonggg committed:
Level-0

zedonggg committed:
Level-1

zedonggg committed:
Level-2

zedonggg committed:
Level-3

zedonggg committed:
Level-4

zedonggg committed:
A-TextUiTesting

zedonggg committed:
Level-5

zedonggg committed:
Level-6

zedonggg committed:
add more oop for task list

zedonggg committed:
Added autosave and load from txt

zedonggg committed:
fix autosave bugs

zedonggg committed:
modified README

zedonggg committed:
add more autosave fixes

zedonggg committed:
Merge branch 'branch-Level-7'

zedonggg committed:
Add LocalDateTime support to tasks

zedonggg committed:
Merge branch 'branch-Level-8'

zedonggg committed:
Add OOP for commands and ui

zedonggg committed:
Merge branch 'A-MoreOOP'

zedonggg committed:
Organize into packages

zedonggg committed:
Merge branch 'Feature-Packages'

zedonggg committed:
Merge branch 'add-gradle-support'

zedonggg committed:
Add some JUnit tests for DeleteCommand and DateTimeParser

zedonggg committed:
Modify gradle config

zedonggg committed:
Add javadoc header comments

zedonggg committed:
Fix styling errors

zedonggg committed:
Add support for searching tasks

zedonggg committed:
Merge Javadoc branch

zedonggg committed:
Merge branch 'branch-A-CodingStandard'

sciphi-123 commented on src/main/java/Aikhsu.java:
```diff
@@ -0,0 +1,46 @@
+import Commands.*;
```
Would it be good for the specific imported libraries be specified rather than a wildcard import?

zedonggg replied:
yes i got lazy

sciphi-123 commented on src/main/java/Aikhsu.java:
```diff
@@ -0,0 +1,46 @@
+        FileHandler.saveTasks(this.tasks, this.filepath);
+        return r;
+    }
+    private void run() {
```
LGTM! Just a minor comment: would it be better to use line breaks to improve readability?

zedonggg replied:
Agreed

sciphi-123 commented on src/main/java/Tasks/Deadline.java:
```diff
@@ -0,0 +1,71 @@
+/**
+ * Represents a deadline task that must be completed by a specific date and time.
+ */
+public class Deadline extends Task {
```
Good use of OOP concepts :)

sciphi-123 reviewed (commented):
Code is complete and well-organised!

sciphi-123 commented on src/main/java/Tasks/Deadline.java:
```diff
@@ -0,0 +1,71 @@
+     * @param by          The due date of the task in string format.
+     * @param dateTime    The due date and time of the task.
+     */
+    public Deadline(String description, boolean isDone, String by, LocalDateTime dateTime) {
```
Good that boolean variables sound like booleans!

zedonggg committed:
Add gradle checkstyle support

zedonggg committed:
Add JavaFX Gui support

zedonggg committed:
Merge branch 'branch-Level-10'

ernestlsy commented on src/main/java/Commands/AddCommand.java:
```diff
@@ -0,0 +1,81 @@
+package Commands;
+
+import Commands.Command;
+import Commands.CommandResult;
+import Exceptions.AikhsuException;
+import Handlers.DateTimeParser;
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Todo;
+import Tasks.TaskList;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
```
```suggestion
import java.time.LocalDateTime;
import java.time.LocalTime;

import Commands.Command;
import Commands.CommandResult;
import Exceptions.AikhsuException;
import Handlers.DateTimeParser;
import Tasks.Deadline;
import Tasks.Event;
import Tasks.Todo;
import Tasks.TaskList;
```
I think imports for java.* should come before all other imports

ernestlsy commented on src/main/java/Commands/Command.java:
```diff
@@ -0,0 +1,30 @@
+public abstract class Command {
+
+    /** The list of tasks that the command will operate on. */
+    protected TaskList tasks;
```
Might be better to name the TaskList as taskList instead of just tasks

ernestlsy commented on src/main/java/Commands/Command.java:
```diff
@@ -0,0 +1,30 @@
+package Commands;
```
Perhaps it would be better to standardize package names to be in fully lowercase
```suggestion
package commands;
```

ernestlsy commented on src/main/java/Handlers/FileHandler.java:
```diff
@@ -0,0 +1,106 @@
+ * This class provides methods to write tasks to a file and retrieve them when needed.
+ */
+public class FileHandler {
+    private static String FILE_PATH;
```
Should this be in camelCase instead? Since it isn't a constant which is immutable.
```suggestion
    private static String filePath;
```

ernestlsy reviewed (commented):
LGTM! Good job ensuring most of the names comply with the naming standards! Just a few nits regarding import order and capitalization.

zedonggg committed:
feat(commands) : add assertions to ensure valid inputs

Added assertions within the Commands and DateParser objects to ensure valid inputs are passed before execution. This helps prevent runtime errors and improves stability.

zedonggg committed:
Refactor code quality

Changed wildcard imports to individual packages and cleaned up redundant code lines.

zedonggg committed:
Merge pull request #2 from zedonggg/branch-A-Assertions

Add assertions to ensure valid inputs

zedonggg committed:
Merge branch 'branch-A-CodeQuality'

zedonggg committed:
Merge pull request #3 from zedonggg/branch-A-CodeQuality

Refactor code quality

zedonggg committed:
Merge branch 'master' of github.com:zedonggg/ip

zedonggg committed:
Add view schedule by date

Added the ability to view deadlines and event on a given date, sorted chronologically. Usage: findbydate DD/MM/YYYY

zedonggg committed:
Merge branch 'branch-B-ViewSchedules'

zedonggg committed:
tidy up code

zedonggg committed:
add more gui tweaks

zedonggg committed:
add final release polishes

zedonggg committed:
update readme
